felix/config/config_params.go

```diff
@@ -331,7 +331,9 @@ type Config struct {
 	// - calicoIPAM: use IPAM data to contruct routes.
 	RouteSource string `config:"oneof(WorkloadIPs,CalicoIPAM);CalicoIPAM"`
 
-	RouteTableRange idalloc.IndexRange `config:"route-table-range;1-250;die-on-fail"`
+	// RouteTableRange is deprecated in favour of RouteTableRanges
+	RouteTableRange  idalloc.IndexRange   `config:"route-table-range;1-250;die-on-fail"`
+	RouteTableRanges []idalloc.IndexRange `config:"route-table-ranges;1-10000;die-on-fail"`
```

Comment:
I think you might have missed the actual defaulting logic for the case where `RouteTableRanges` is not set.  I'd add a method to the ConfigParams object that checks which of the two fields is set explicitly and returns an `[]idalloc.IndexRange` in both cases.